tests/unit/controlgroup/non-empty-class-key.js

```diff
@@ -0,0 +1,51 @@
+define( [
+	"qunit",
+	"jquery",
+	"ui/widgets/controlgroup",
+	"ui/widgets/selectmenu"
+], function( QUnit, $ ) {
+
+QUnit.module( "Non-empty class key", {
+	setup: function() {
+		this.classKey = $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ];
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] =
+			$.trim( $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] +
+				" something-custom" );
+	},
+	teardown: function() {
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] = this.classKey;
+	}
+} );
+
+QUnit.test( "Controlgroup handles non-empty class key correctly", function( assert ) {
+	expect( 2 );
+	var select, resultingClasses;
+	var group = $( ".controlgroup-non-empty-class-key" )
+		.controlgroup()
+		.controlgroup( "instance" );
+
+	select = $( "<select><option>Option 1</option><option>Option 2</option></select>" )
+		.appendTo( group.element )
+		.selectmenu()
+		.selectmenu( "instance" );
+	group.refresh();
+	assert.deepEqual(
+		( ( ( select.options.classes[ "ui-selectmenu-button-closed" ] || "" )
+			.match( /\S+/g ) ) || [] ).sort(),
+		[ "something-custom", "ui-corner-all" ],
+		"Adding one item to a controlgroup results in the expected value for the class key" );
```

Comment:
Why aren't you just doing `hasClasses( select, "something-custom ui-corner-all" );` ?